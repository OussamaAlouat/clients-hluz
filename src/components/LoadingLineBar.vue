<template>
  <v-progress-linear
    height="10"
    striped
    color="deep-orange"
    v-model="value"
    :active="display"
    :indeterminate="query"
    :query="true"
  >
  </v-progress-linear>
</template>

<script>
export default {
  name: 'LoadingLineBar',
  props: {
    isLoaded: {
      type: Boolean,
      required: true,
    },
  },

  data() {
    return {
      value: 0,
      query: false,
      display: true,
      interval: 0,
      increment: 45,
    };
  },

  mounted() {
    this.queryAndIndeterminate();
  },

  beforeDestroy() {
    clearInterval(this.interval);
  },

  methods: {
    queryAndIndeterminate() {
      this.query = true;
      this.display = true;
      this.value = 0;

      setTimeout(() => {
        this.query = false;

        // eslint-disable-next-line consistent-return
        this.interval = setInterval(() => {
          if (this.value === 100) {
            clearInterval(this.interval);
            this.display = false;
            return setTimeout(this.queryAndIndeterminate, 2000);
          }

          this.value += this.incremet;
        }, 1000);
      }, 2500);
    },
  },

  watch: {
    isLoaded() {
      if (this.isLoaded) {
        this.increment = 100;
      }
    },
  },
};
</script>

<style>

</style>
