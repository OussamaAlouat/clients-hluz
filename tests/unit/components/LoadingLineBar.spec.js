import { shallowMount } from '@vue/test-utils';
import LoadingLineBar from '@/components/LoadingLineBar.vue';

describe('LoadingLineBar.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(LoadingLineBar, {
      propsData: {
        isLoaded: false,
      },
    });
  });

  it('Component render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
