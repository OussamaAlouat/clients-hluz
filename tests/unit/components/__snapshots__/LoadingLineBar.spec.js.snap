// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingLineBar.vue Component render 1`] = `<v-progress-linear-stub color="deep-orange" value="0" active="true" buffervalue="100" height="10" query="true" striped="true" indeterminate="true"></v-progress-linear-stub>`;
